mir_eval/sonify.py

```diff
@@ -147,34 +154,42 @@ def __interpolator(x):
             return value
         return __interpolator
 
+    # print(f"sonify.time_frequency length {length}, n_times {n_times}, gram shape {gram.shape}, times shape {times.shape}")
+
     # Threshold the tfgram to remove non-positive values
     gram = np.maximum(gram, 0)
 
     # Pre-allocate output signal
     output = np.zeros(length)
     time_centers = np.mean(times, axis=1) * float(fs)
 
+    # Not really a true spectral energy, just a summation for optimisation.
+    spectral_sums = np.sum(gram, axis = 1)
     for n, frequency in enumerate(frequencies):
+        if spectral_sums[n] < 0.01: # TODO set threshold intelligently.
+            continue
         # Get a waveform of length samples at this frequency
         wave = _fast_synthesize(frequency)
+        # print(f"Freq {n} {frequency}Hz energy {spectral_sums[n]}, length of wave {len(wave)}")
```

Comment:
Please remove commented debug prints

Reply:
Removing diagnostics removes the ability to build on the expected semantics reflected in the diagnostic messages to aid future debugging. Effectively each developer has to begin diagnostics and understand the code from scratch. But I've removed them in answer to your request.